client/components/recipes.js

```diff
@@ -21,11 +43,11 @@ class Recipes extends React.Component {
           <Link to="submit" className="waves-effect waves-light btn"><i className="material-icons left">local_dining</i>Add Your Own!</Link>
         </div>
         <div className="recipe-entries">
-          <RecipeEntry />
-          <RecipeEntry />
-          <RecipeEntry />
-          <RecipeEntry />
-          <RecipeEntry />
+          {
+            (!this.state.recipes.length)
+              ? <span> Loading Recipes... </span>
```

Comment:
Since recipes starts off as an empty array, won't this display 'Loading Recipes' even before the user initiates a search? That seems a little confusing from a ui perspective.  Can you have it display something else (or nothing) initially and only display 'loading' while waiting for the api call to resolve?

Reply:
Just to clarify, we don't want to show any recipes before a user initiates search? So just a blank page with a search bar? Right now getAllRecipes is invoked with componentDidMount, and the loading message is just a safeguard in case the db query takes a bit too long.

Reply:
If a user is going to the Recipes page to look for specific recipes, is it beneficial to start out providing them with a list that may or may not include what they're looking for? I had thought we would only populate the page on search.